scss/_teamshares.scss

```diff
@@ -198,292 +198,294 @@ ul a {
 // TODO: Refactor all ts-button-primary to ts-button primary and remove from this styling
 button,
 .ts-button {
-  appearance: none;
-  border: 0;
-  border-radius: 0.375rem;
-  cursor: pointer;
-  display: inline-block;
-  font-family: theme('fontFamily.sans');
-  font-size: 0.8125rem;
-  font-weight: $font-weight-medium;
-  letter-spacing: -0.01rem;
-  line-height: 2rem;
-  min-width: 5rem;
-  padding: 0.15rem 1rem;
-  text-align: center;
-
-  // We need to use custom styling when we use sizing (sm, md etc.) for links. If we leave this in ts-button it messes up existing button styles.
-  &.sized-link {
-    align-items: center;
-    display: flex;
-    justify-content: center;
-    padding: 0 !important;
-  }
-
-  &.blue {
-    @apply bg-blue-700 text-white hover:text-white;
-  }
-
-  &.xs {
-    font-size: 0.75rem;
-    line-height: 1rem;
-    min-width: 0;
-    padding: 0;
-
-    &:hover {
-      text-decoration: underline;
-    }
-  }
-
-  &.sm {
-    font-size: 0.8125rem;
-    line-height: 1.9375rem;
-    padding: 0 0.5rem;
-  }
-
-  &.md {
+  :not(.button-view-component, .button-icon, .icon-button-view-component) {
+    appearance: none;
+    border: 0;
+    border-radius: 0.375rem;
+    cursor: pointer;
+    display: inline-block;
+    font-family: theme('fontFamily.sans');
     font-size: 0.8125rem;
-    line-height: 1.9375rem;
+    font-weight: $font-weight-medium;
+    letter-spacing: -0.01rem;
+    line-height: 2rem;
+    min-width: 5rem;
     padding: 0.15rem 1rem;
-  }
-
-  &.lg {
-    font-size: 0.875rem;
-    line-height: 1.9375rem;
-    min-width: 10rem;
-    padding: 0.15rem 2rem;
     text-align: center;
-  }
 
-  &.lg.icon {
-    padding-left: 2rem;
+    // We need to use custom styling when we use sizing (sm, md etc.) for links. If we leave this in ts-button it messes up existing button styles.
+    &.sized-link {
+      align-items: center;
+      display: flex;
+      justify-content: center;
```

Comment:
Question on how the design group is approaching styles like these -- it COULD be rewritten to use tailwind classes, which would then make it clear which styles are custom overrides, but I'm not sure which approach is preferable.

e.g. these three lines _could_ be (not sure if they _should_ be) replaced with `@apply flex justify-center items-center`

Reply:
We're going to end up deprecating all the existing styles in this `button,
.ts-button` styling chunk as we create link/button view components, the only changes made to this styling was to declare that existing button styles should not be applied to elements with our new view component classes below